src/test/scala/com/databricks/labs/mosaic/test/MosaicSpatialQueryTest.scala

```diff
@@ -0,0 +1,136 @@
+            super.test(testName + s" (codegen) (${geom.name}, ${is.name})", testTags: _*)(
+              withSQLConf(
+                SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "false",
+                SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "true",
+                SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.CODEGEN_ONLY.toString
+              ) {
+                  withMosaicConf(geom, is) {
+                      testFun
+                  }
+              }
+            )
+
+        }
+    }
+
+    protected def testAllGeometriesNoCodegen(testName: String, testTags: Tag*)(testFun: MosaicContext => Unit): Unit = {
+        val is = MockIndexSystem
+        for (geom <- geometryApis) {
+            super.test(testName + s" (no codegen) (${geom.name}, ${is.name})", testTags: _*)(
+              withSQLConf(
+                SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "false",
+                SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.NO_CODEGEN.toString
+              ) {
+                  withMosaicConf(geom, is) {
+                      testFun
+                  }
+              }
+            )
+        }
+    }
+
+    protected def testAllCodegen(testName: String, testTags: Tag*)(testFun: MosaicContext => Unit): Unit = {
+        for (geom <- geometryApis) {
+            for (is <- indexSystems) {
+                super.test(testName + s" (codegen) (${geom.name}, ${is.name})", testTags: _*)(
+                  withSQLConf(
+                    SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "false",
+                    SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "true",
+                    SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.CODEGEN_ONLY.toString
+                  ) {
+                      withMosaicConf(geom, is) {
+                          testFun
+                      }
+                  }
+                )
+            }
+        }
+    }
+
+    protected def testAllNoCodegen(testName: String, testTags: Tag*)(testFun: MosaicContext => Unit): Unit = {
+        for (geom <- geometryApis) {
+            for (is <- indexSystems) {
+                super.test(testName + s" (no codegen) (${geom.name}, ${is.name})", testTags: _*)(
+                  withSQLConf(
+                    SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "false",
+                    SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.NO_CODEGEN.toString
+                  ) {
+                      withMosaicConf(geom, is) {
+                          testFun
+                      }
+                  }
+                )
+            }
+        }
+    }
+
+    def checkGeometryTopo(
+        mc: MosaicContext,
+        actualAnswer: DataFrame,
+        expectedAnswer: DataFrame,
+        geometryFieldName: String
+    ): Unit = {
+        MosaicSpatialQueryTest.checkGeometryTopo(mc, actualAnswer, expectedAnswer, geometryFieldName)
+    }
+
+}
+
+object MosaicSpatialQueryTest extends Assertions {
```

Comment:
This object should be in a separate file since it is not following "companion object" naming convention.

Reply:
Can we also add some scaladocs?
I apreciate this is test package but these contain testing APIs so should have some scaladocs to make it easier for others to use.

Reply:
I'm still a huge scala noob, so it would be greatly appreciated if you could point me to some resources explaining this "companion object naming convention". I used essentially the same approach as spark (https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala)

Reply:
I must have misread it earlier - names of object and class should be the same if in same file :) but I see that is the case